File: selenium/pages/homePage.js
const { BasePage } = require("./basePage");

class HomePage extends BasePage {
  async assertLoaded() {
    await this.assertHasText("Welcome to Smart Furnace AI");
    await this.assertHasText("Home");
  }

  async openTab(tabName) {
    await this.tapText(tabName);
  }
}

module.exports = { HomePage };

File: appium/pages/home_page.py
from pages.base_page import BasePage


class HomePage(BasePage):
    def assert_loaded(self) -> None:
        assert self.has_text("Welcome to Smart Furnace AI")
        assert self.has_text("Home")

    def open_tab(self, tab_name: str) -> None:
        self.tap_text(tab_name)
